src/wallet/rpc/transactions.cpp

```diff
@@ -22,16 +22,23 @@ static void WalletTxToJSON(const CWallet& wallet, const CWalletTx& wtx, UniValue
     entry.pushKV("confirmations", confirms);
     if (wtx.IsCoinBase())
         entry.pushKV("generated", true);
-    if (auto* conf = wtx.state<TxStateConfirmed>())
-    {
+    if (auto* conf = wtx.state<TxStateConfirmed>()) {
         entry.pushKV("blockhash", conf->confirmed_block_hash.GetHex());
         entry.pushKV("blockheight", conf->confirmed_block_height);
         entry.pushKV("blockindex", conf->position_in_block);
         int64_t block_time;
         CHECK_NONFATAL(chain.findBlock(conf->confirmed_block_hash, FoundBlock().time(block_time)));
         entry.pushKV("blocktime", block_time);
+        if (chain.hasAssumedValidChain()) {
+            entry.pushKV("assumed", wallet.IsTxAssumed(wtx));
+        }
     } else {
         entry.pushKV("trusted", CachedTxIsTrusted(wallet, wtx));
+        // This can happen when loading a wallet from another node,
+        // while still in (background) IBD:
```

Comment:
82f19c7e97281bbd66a25e53f30fe2f80a4f9d11: I found this while loading an existing wallet into a fresh (signet) node, while it was loading doing the background sync. That's not a very common scenario, so I haven't bothered looking into deeper causes...